### PR TITLE
Playlist

### PR DESCRIPTION
**Pull request by atharvabhanage02**
This PR has functionalities

- View History
- Delete video from History
- Clear History
- Create Playlist
- Add video to Playlist
- Remove video from Playlist
- Delete particular Playlist

Files to look for : 

- History Context : `src/Context/HistoryContext`
- Playlist Context : `src/Context/PlaylistContext`
- History Page : `src/Pages/History`
- Playlist Reducer : `src/Reducers/playlist-reducer.js`
- Playlist Page : `src/Pages/Playlist`
- PlaylistSingle Page : `src/Pages/PlaylistSinglePage`

Deploy Preview : https://deploy-preview-5--cricplay.netlify.app/

### CODE REVIEW

**Review by yashraj2312 — APPROVED** [review]
Looks good to merge

File: src/Components/Playlist-Option/PlaylistOption.jsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+              className={`login-btn white-border ${
+                playlistInfo.title === "" || playlistInfo.description === ""
+                  ? "disable-btn"
+                  : ""
+              }`}
+              onClick={() => {
+                createPlaylistFunc();
+                setPlaylistInfo({ title: "", description: "" });
+                // setTooglePlaylistOption((prev) => !prev);
+              }}
```
Comment (anjali1102) [inline src/Components/Playlist-Option/PlaylistOption.jsx]: can you remove this

File: src/Pages/Playlist/Playlist.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        ) : (
+          playlists.map((video) => {
+            return (
+              <div className="video-card border-1p">
+                <div>
+                  <Link to={`/playlist/${video._id}`}>
+                    <img
+                      className="video-thumbnail"
+                      src="assets/Thumbnail/video-1.webp"
+                      alt=""
```
Comment (anjali1102) [inline src/Pages/Playlist/Playlist.jsx]: alt is empty

**Review by anjali1102 — APPROVED** [review]
Functionality working fine. You can remove console.log coming in the deployed link. You can stack all error code with console.error(" ").
image

**Comment by khushijohri2001** [comment]
Looks clean to me. Resolve the mentioned issues and the good to merge

File: src/App.js [hunk 3]
```diff
@@ -8,6 +8,9 @@ import { WatchLaterPage } from "./Pages/Watch-Later-Page/WatchLaterPage";
 import { RequiresAuth } from "./Components/RequiresAuth";
 import Mockman from "mockman-js";
 import { LikePage } from "./Pages/Like-Page/LikePage";
+import { History } from "./Pages/History/History";
+import { Playlist } from "./Pages/Playlist/Playlist";
+import { PlaylistSinglePage } from "./Pages/PlaylistSinglePage/PlaylistSinglePage";
```
Comment (HeyNitin) [inline src/App.js:13]: You can have a look at absolute imports, try them once. just a suggestion

File: src/App.js [hunk 4]
```diff
@@ -16,6 +19,14 @@ function App() {
         <Route path="/" element={<Explore />} />
         <Route path="/login" element={<Login />} />
         <Route path="/signup" element={<Signup />} />
+        <Route
+          path="/history"
+          element={
+            <RequiresAuth>
+              <History />
+            </RequiresAuth>
+          }
```
Comment (HeyNitin) [inline src/App.js:28]: Instead of having routes in App, you can have a <Route /> component

File: src/Pages/Playlist/Playlist.jsx (hunk 2 above)
Comment (HeyNitin) [inline src/Pages/Playlist/Playlist.jsx]: add "thumbnail in alt of the image tag, leaving it empty is not a good practice"

**Review by HeyNitin — APPROVED** [review]
Left some suggestions, rest look good. you can merge after deleting commented code

File: src/Components/Playlist-Option/PlaylistOption.jsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
+import "./playlist-modal.css";
+import { AiOutlineClose } from "../../Utils/icons";
+import { useState } from "react";
+import { usePlaylist } from "../../Context/PlaylistContext/PlaylistContext";
```
Comment (vashnavichauhan1825) [inline src/Components/Playlist-Option/PlaylistOption.jsx:4]: suggestion:-use absolute path

File: src/Pages/Playlist/Playlist.jsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  const deletePlaylistFunc = async (id) => {
+    try {
+      const data = await deletePlaylist(id, auth.tokenValue);
+      dispatchPlaylist({ type: "DELETE_PLAYLIST", payload: data.playlists });
+    } catch (error) {
+      console.log("error in deleting the whole playlist function", error);
+    }
+  };
+  return (
+    <div>
```
Comment (vashnavichauhan1825) [inline src/Pages/Playlist/Playlist.jsx]: suggestion:- use Fragment will not add any additional elements into the DOM tree, whereas, using a div will add a div to the DOM tree.

**Review by vashnavichauhan1825 — APPROVED** [review]
just few suggestions otherwise ,everything looks goot to me

File: src/Components/Playlist-Option/PlaylistOption.jsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  addVideoToPlaylist,
+  createPlaylist,
+  deletePlaylist,
+  deleteVideoFromPlaylist,
+  getPlaylists,
+} from "../../Services/playlistServices";
+import { useAuth } from "../../Context/Auth/auth-context";
+import { useVideos } from "../../Context/VideosContext/VideosContext";
+const PlayistOption = ({ videoData }) => {
+  // const [playlists, setPlaylists] = useState(["Amazing T20"]);
```
Comment (parulgupta2001) [inline src/Components/Playlist-Option/PlaylistOption.jsx]: remove commented code, if not required

**Review by therajatg — APPROVED** [review]
Remove console.log from code. Everything else looks good.

File: src/Components/Playlist-Option/PlaylistOption.jsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+  const createPlaylistFunc = async () => {
+    try {
+      const data = await createPlaylist(playlistInfo, auth.tokenValue);
+      dispatchPlaylist({
+        type: "CREATE_NEW_PLAYLIST",
+        payload: data.playlists,
+      });
+    } catch (error) {
+      console.log("error in creating playlist in main component", error);
```
Comment (parulgupta2001) [inline src/Components/Playlist-Option/PlaylistOption.jsx:34]: instead of consoling error, show error via toast to the user

**Review by parulgupta2001 — APPROVED** [review]
some changes are suggested otherwise everything looks good

**Pull request merged**